### DISCUSSION TITLE
THEME: Contour lines not visible

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1014] [Q&A] [answered]
I downloaded the default.xml theme, saved in assets folder and called:

```
// Render theme
AssetsRenderTheme assetsRenderTheme = new AssetsRenderTheme(requireContext().getAssets(), "", THEME_DEFAULT);
theme = map.setTheme(assetsRenderTheme);
```

until here it's ok.

Now, for test, I edited the default.xml theme, adding the contour lines (found as an answer in a post):
```
<!-- contour lines -->
<m e="way" k="contour_ext">
    <m v="elevation_minor">
        <m zoom-max="13" zoom-min="13">
            <line stroke="#641616" width="0.4" />
        </m>
    </m>
    <m v="elevation_medium|elevation_major">
        <m zoom-max="13" zoom-min="13">
            <line stroke="#630a0c" width="0.6" />
        </m>
    </m>
</m>
```

No contour lines drawed...

Where can be the problem? wrong key "contour_ext" ? wrong values "elevation_minor"? 

I use mapsforge map V5.

**Comment by devemux86** [comment, marked as answer]
Map data must be present both in the map file and in the theme to be rendered.

- [OpenAndroMaps](https://www.openandromaps.org/en/) maps contain contour lines (with [Elevate](https://www.openandromaps.org/en/legend/elevate-mountain-hike-theme) and [Voluntary](https://voluntary.nichesite.org) themes)
- [Freizeitkarte](https://www.freizeitkarte-osm.de/android/en/index.html) maps contain contour lines (with [Freizeitkarte](https://www.freizeitkarte-osm.de/android/en/design.html) themes)
- [Mapsforge](https://download.mapsforge.org/) maps do not contain contour lines

**Comment by moiatto** [comment]
I was convinced that maps had it
Understood. Thank you.
